Firestore/core/src/firebase/firestore/remote/remote_objc_bridge.mm

```diff
@@ -306,22 +298,21 @@
     const std::vector<grpc::ByteBuffer>& responses) const {
   // Sort by key.
   std::map<DocumentKey, MaybeDocument> results;
-  nanopb::Reader reader;
 
   for (const auto& response : responses) {
-    auto maybe_proto =
+    GrpcByteBufferReader grpc_reader{response};
+    auto message =
         Message<google_firestore_v1_BatchGetDocumentsResponse>::TryParse(
-            google_firestore_v1_BatchGetDocumentsResponse_fields, response);
-    if (!maybe_proto.ok()) {
-      return maybe_proto.status();
-    }
+            &grpc_reader);
 
-    const auto& proto = maybe_proto.ValueOrDie();
-    MaybeDocument doc = serializer_.DecodeMaybeDocument(&reader, *proto);
+    nanopb::Reader reader;
```

Comment:
I suppose I could turn it into something like `nanopb::Reader reader = grpc_reader.ReleaseReader();`, which would preserve a failed status.

Reply:
If `GrpcByteBufferReader` is-a `Reader` it would be usable for this purpose here too.